### PR TITLE
test: move "01"-dependent tests

### PR DESCRIPTION
**Pull request by vindard**
_Merge after #3677_

## Description

This PR moves the following test suites to new integration structure:
- `invoices.spec.ts`
- `parse-errors.spec.ts`

These were chosen because they still had lingering dependencies on the `01` setup folder tests

### CODE REVIEW

File: core/api/test/integration/services/lnd-service.spec.ts [hunk 1]
```diff
@@ -209,160 +214,281 @@ afterAll(async () => {
...
-    expect(paid).toBeInstanceOf(PaymentRejectedByDestinationError)
-  })
 
-  it("pay invoice with High CLTV Delta", async () => {
-    // Create invoice
-    const { request } = await createInvoice({
-      lnd: lndOutside1,
-      tokens: amountInvoice,
-      cltv_delta: 200,
+    it("probes across route with fee higher than payment amount", async () => {
```
Comment (dolcalmi) [inline core/api/test/integration/services/lnd-service.spec.ts:309]: probably is better `it("fails` but not a blocker

File: core/api/test/integration/services/lnd-service.spec.ts [hunk 2]
```diff
@@ -209,160 +214,281 @@ afterAll(async () => {
...
+        decodedInvoice: lnInvoice,
+        btcPaymentAmount,
+        maxFeeAmount: undefined,
+      })
+      if (paid instanceof Error) throw paid
+
+      // Confirm payment exists in lnd
+      const retrievedPayment = await lndService.lookupPayment({ paymentHash })
+      expect(retrievedPayment).not.toBeInstanceOf(Error)
+      if (retrievedPayment instanceof Error) return retrievedPayment
```
Comment (dolcalmi) [inline core/api/test/integration/services/lnd-service.spec.ts:353]: this must be a throw

File: core/api/test/integration/services/lnd-service.spec.ts [hunk 3]
```diff
@@ -209,160 +214,281 @@ afterAll(async () => {
...
+      expect(retrievedPayment.status).toBe(PaymentStatus.Settled)
+      if (retrievedPayment.status !== PaymentStatus.Settled) return
```
Comment (dolcalmi) [inline core/api/test/integration/services/lnd-service.spec.ts:355]: must be a throw?

File: core/api/test/legacy-integration/app/wallets/invoice.spec.ts
Comment (dolcalmi) [inline core/api/test/legacy-integration/app/wallets/invoice.spec.ts:1]: did you only move rate limits validations? are the other tests in bats/other files?

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: core/api/test/legacy-integration/app/wallets/invoice.spec.ts
Comment (vindard) [reply core/api/test/legacy-integration/app/wallets/invoice.spec.ts:1]: Yup these should be covered in bats already https://github.com/GaloyMoney/galoy/pull/3694/commits/232173ca4b3fe162da511641c9ef12c67626e9f1

File: core/api/test/integration/services/lnd-service.spec.ts (hunk 3 above)
Comment (vindard) [reply core/api/test/integration/services/lnd-service.spec.ts]: https://github.com/GaloyMoney/galoy/pull/3694/commits/dbe2248c4049bc54d967e6c1735eb815ba905dcb

File: core/api/test/integration/services/lnd-service.spec.ts (hunk 2 above)
Comment (vindard) [reply core/api/test/integration/services/lnd-service.spec.ts]: https://github.com/GaloyMoney/galoy/pull/3694/commits/dbe2248c4049bc54d967e6c1735eb815ba905dcb

File: core/api/test/integration/services/lnd-service.spec.ts (hunk 1 above)
Comment (vindard) [reply core/api/test/integration/services/lnd-service.spec.ts]: https://github.com/GaloyMoney/galoy/pull/3694/commits/dbe2248c4049bc54d967e6c1735eb815ba905dcb

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**